pandas/tests/extension/base/reshaping.py

```diff
@@ -170,3 +170,25 @@ def test_merge(self, data, na_value):
                  [data[0], data[0], data[1], data[2], na_value],
                  dtype=data.dtype)})
         self.assert_frame_equal(res, exp[['ext', 'int1', 'key', 'int2']])
+
+    @pytest.mark.parametrize("columns", [
+        ["A", "B"],
+        pd.MultiIndex.from_tuples([('A', 'a'), ('A', 'b')],
+                                  names=['outer', 'inner']),
+    ])
+    def test_stack(self, data, columns):
+        df = pd.DataFrame({"A": data[:5], "B": data[:5]})
+        df.columns = columns
+        result = df.stack()
+        expected = df.astype(object).stack()
+        # we need a second astype(object), in case the constructor inferred
+        # object -> specialized, as is done for period.
+        expected = expected.astype(object)
```

Comment:
This is kinda strange. For `DataFrame[ndarray[object]].stack()` of all periods, we actually infer `period`-dtype. Do we want that, or should we explicitly pass `dtype=object` when creating the new series / frame to ensure that we don't infer the "correct" dtype?

```python
In [1]: import pandas as pd

In [2]: a = pd.core.arrays.period_array(['2000', '2001'], freq='D')

In [3]: pd.DataFrame({"A": a, "B": a}).astype(object).dtypes
Out[3]:
A    object
B    object
dtype: object

In [4]: pd.DataFrame({"A": a, "B": a}).astype(object).stack().dtype
Out[4]: period[D]
```

(that's on master)